### DISCUSSION TITLE
cakephp autoinstall and 8.3 version

### DISCUSSION THREAD

**Discussion by viniciusbig** [shinsenter/php#70] [General]
So, I'm using the cakephp4 image for a while. 
Really nice job.

But now, Cakephp5 is out and I'm the process of upgrading everything.
I'm taking a look into all images in v5 here, trying to understand the main differences.

For now, I have 2 questions:

#1 
Were the installment scripts removed on v5? If so, why?
I couldn't find the cakephp or the laravel install script for example.
Nothing happens when starting this image since there is no script on `rootfs/startup`

It seems that, right now, most of "webapp" images are just regular php images.
Am I missing something?

#2 
Why cakephp4:8.3 is deprecated?
Is there a plan to create the 8.3 tag?
(the main php package has a 8.3, right?)

Appreciate any feedback

**Comment by shinsenter** [comment]
Sure, here's a more polished translation:

---

Hello @viniciusbig,

Thank you for reaching out. Here are the answers to your questions:

1. At the moment, I don't have any Docker images that support CakePHP version 5. Therefore, there are no installation scripts available for CakePHP v5 in the Docker images I've released. However, I believe the existing Docker images for CakePHP might still be suitable for running CakePHP v5 with your current project.

2. According to the [official CakePHP website](https://book.cakephp.org/4/en/installation.html#), version 4 only supports up to PHP 8.2. Therefore, there is no Docker image with the tag `shinsenter/cakephp4:php8.3`.

We will work on adding Docker images that support CakePHP v5 as soon as possible, but it may take some time.

Thank you for your understanding and patience.

**Comment by shinsenter** [comment]
Sorry for sending the wrong message from a different thread earlier.

**Comment by shinsenter** [comment]
Hi @viniciusbig 

Docker images for CakePHP5 are now available here.
https://hub.docker.com/r/shinsenter/cakephp5/tags

**Comment by viniciusbig** [comment]
Thanks for that!

Like you said, right now cakephp4 image can handle a cakephp5 installation just fine.
But its nice to have the correct version to use.
I'm exploring you images and testing a lot.

I'm starting to understand some changes between version 4 and 5. (again, I was using version 4 for a while)

So, I'd like to highlight something I discovered for anyone else wondering, and check with you if I got this right.

I've just notice you don't have a install script for projects anymore (on `rootfs/startup`) and instead, a new environment variable `INITIAL_PROJECT` will do the job if the folder is empty. (and that is a really nice touch)

So, this varibale is used to pre-install a project using a `composer create-project` command, right?
It could be a laravel project with `INITIAL_PROJECT = laravel/laravel`
or even a cakephp project with `INITIAL_PROJECT = cakephp/app`
or, to be more version specific, `INITIAL_PROJECT = cakephp/app:~4.0` will install cakephp4 from scratch.

But any other project that starts with `composer create-project` can use this feature, right?

I'm wondering if you have a comparison between v4 and v5.


either way... I think version 5 is way simpler to use

**Reply by shinsenter** [reply]
@viniciusbig 

You are totally correct. The environment variable `INITIAL_PROJECT` will tell the container which project to install when it starts.

Currently, the Docker setups for CakePHP4 and CakePHP5 differ only in the version of CakePHP - which is specified in the `INITIAL_PROJECT` environment variable. The other components related to the runtime environment are quite similar (they both use the `app` directory for the application, the `cake` command in the `bin` directory, etc.).

Additionally, if you look closely at the tags, the Docker images for CakePHP4 only support PHP up to version 8.2, while the Docker images for CakePHP5 are available for PHP versions 8.1 to 8.3.
